airflow/utils/json.py

```diff
@@ -123,3 +136,89 @@ def dumps(self, obj, **kwargs):
 
     def loads(self, s: str | bytes, **kwargs):
         return json.loads(s, **kwargs)
+
+
+# for now separate as AirflowJsonEncoder is non-standard
+class XComEncoder(json.JSONEncoder):
+    """This encoder allows serializes any object that has attr."""
+
+    def default(self, o: object) -> dict:
+        from airflow.serialization.serialized_objects import BaseSerialization
+
+        if hasattr(o, "serialize"):
+            classname = o.__module__ + "." + o.__class__.__name__
+            version = getattr(o.__class__, "version", DEFAULT_VERSION)
+            return {
+                CLASSNAME: classname,
+                VERSION: version,
+                DATA: getattr(o.__class__, "serialize")(o)
```

Comment:
Shouldn't this be

```suggestion
                DATA: getattr(o, "serialize")()
```

Reply:
That's not what stackoverflow is saying https://stackoverflow.com/a/38764817 ... but maybe you are right for `serialize` as it is an instance method. `deserialize` is a staticmethod